pkg/rules/rulespb/custom.go

```diff
@@ -41,12 +43,140 @@ func NewRecordingRule(r *RecordingRule) *Rule {
 	}
 }
 
+// Cmp compares equal recording rules r1 and r2 and returns:
+//
+//   < 0 if r1 < r2  if rule r1 is lexically before rule r2
+//     0 if r1 == r2
+//   > 0 if r1 > r2  if rule r1 is lexically after rule r2
+//
+// More formally, the ordering is determined in the following order:
+//
+// 1. recording rule last evaluation (earlier evaluation comes first)
+//
+// Note: This method assumes r1 and r2 are logically equal as per Rule#Cmp.
+func (r1 *RecordingRule) Cmp(r2 *RecordingRule) int {
```

Comment:
```suggestion
func (r1 *RecordingRule) Compare(r2 *RecordingRule) int {
```
This is for consistency with other Comparators in Thanos.

BTW it feels super weird. Because it compares nothing really (: Just single field. I think we might want to either have some quick function or something, as it can only confuse user of `Compare` if it does not Compare, just checks `LastEval` :thinking:

Reply:
Good point, I think when I coded this I first implemented the method outer shell and then filled it with life which turned out to be just comparing `LastEval` :-)

> Why not following pattern visible for other structs? 

Which pattern do you refer too? I aligned this with other `Compare` methods but can refactor.

Reply:
Let's speak offline, not a blocker